test/metrics/metrics_test.go

```diff
@@ -71,10 +75,8 @@ func TestMetric_Counter(t *testing.T) {
 		// Test the metrics for the actions Counter
 		for _, tt := range testUnit {
 			t.Run(tt.name, func(t *testing.T) {
-				counter := tt.c
-				counter.Inc()
 				// Compare the metrics
-				if err := testutil.CollectAndCompare(counter, strings.NewReader(tt.data+tt.nameMetric+tt.value), tt.nameMetric); err != nil {
+				if err := testutil.CollectAndCompare(tt.c, strings.NewReader(tt.data+tt.nameMetric+tt.value), tt.nameMetric); err != nil {
```

Comment:
```suggestion
				if err := testutil.CollectAndCompare(counter, strings.NewReader(tt.data+tt.nameMetric+tt.value), tt.nameMetric); err != nil {
```
In logic, we need to compare the object that we have increment before.